protocol/directory.go

```diff
@@ -70,13 +74,13 @@ func (d *ConiksDirectory) Register(req *RegistrationRequest) (
 		return nil, ErrorDirectory
 	}
 	if bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
-		return NewRegistrationProof(ap, d.LatestSTR(), nil), ErrorNameExisted
+		return NewRegistrationProof(ap, d.LatestSTR(), nil, ErrorNameExisted)
```

Comment:
Hmm ok. It seemed to me like `ErrorDirectory` happens when the `pad.Lookup` returns an error, right, which would indicate an internal server error, right? We should definitely return an error about this to the client, but I think we should treat such an error differently on the server side.